src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/Test/Validator.cs

```diff
@@ -65,8 +65,18 @@ public async Task<ValidationResult> ValidateAsync(ValidationContext context)
         /// <summary>
         /// Checks that the current batch of catalog entries contains the entry that was created from the current state of the database.
         /// </summary>
+        /// <remarks>
+        /// Our validations depend on the fact that the database and V3 are expected to have the same version of a package.
+        /// If the catalog entry we're running validations on, which is supposed to represent the current state of V3, is less recent than the database, then we shouldn't run validations.
+        /// </remarks>
         protected virtual async Task<bool> ShouldRunAsync(ValidationContext context)
         {
+            if (context.Entries == null)
```

Comment:
Looked into adding unit tests around this. Apparently there are none currently and adding some is not trivial. I would like to punt this for now.